src/components/ProductBacklogListComponent.js

```diff
@@ -44,71 +48,121 @@ const ProductBacklogListComponent = () => {
     subtasks: [],
     estimate: 0,
     description: '',
+    estimate: 0,
   };
 
   const [newStory, setNewStory] = useReducer(
     (newStory, newState) => ({ ...newStory, ...newState }),
     initialNewStory
   );
 
-  //Fetch the product backlog for the current project
-//use effect
+  useEffect(async () => {
+    if (project) {
+      const { productBacklog } = project;
+      await setState({ productBacklog });
+      console.log('loaded backlog');
+    } else displayPopup('Project not loaded');
+  }, []);
 
-  const addNewStory = (story) => {
+  useEffect(async () => {
+    if (state.productBacklog) {
+      console.log(state.productBacklog);
+      console.log(project);
+      await updateProject();
+    }
+  }, [state.productBacklog]);
+
+  const addNewStory = async (story) => {
     setState({ productBacklog: [...state.productBacklog, story] });
+    //await updateProject()
+  };
+
+  const updateProject = async () => {
+    //Update the db
+    try {
+      let result = await db.updateProject({
+        projectName: project.projectName,
+        productBacklog: state.productBacklog,
+      });
+
+      //Check if the update worked
+      if (!result.success) throw { message: result.message };
+
+      await refreshProjects(); //Refresh the projects stored as they were modified
+      displayPopup('Successfully saved new user story');
+    } catch (err) {
+      displayPopup(`Error saving user story ${err.message}`);
+
+      console.log(`Project update failed: ${err}`);
+    }
   };
 
-  const onDeleteItem = (item) => {
-    setState({
+  const onDeleteItem = async (item) => {
+    await setState({
       productBacklog: state.productBacklog.filter((story) => story !== item),
     });
+
```

Comment:
setState isn't async is it?

Reply:
oop left that in there. Thought for a while it returned a promise but it is just its own thing. Its fixed now